File: app/src/main/java/com/datarockets/mnchkn/activities/players/PlayersListInteractor.java
package com.datarockets.mnchkn.activities.players;

import com.datarockets.mnchkn.models.Player;

import java.util.List;

public interface PlayersListInteractor {

    interface OnFinishedListener {
        void onPlayersLoaded(List<Player> players);
        void onPlayerAdded(Player player);
        void onPlayerDeleted(int position);
        void onPlayersCountChecked(boolean enough);
        void onGameStarted(boolean started);
    }

    void clearPlayersStats(OnFinishedListener listener);
    void isGameStarted(OnFinishedListener listener);
    void checkIsEnoughPlayer(OnFinishedListener listener);
    void getPlayers(OnFinishedListener listener);
    void addPlayer(String name, OnFinishedListener listener);
    void deletePlayer(int position, long id, OnFinishedListener listener);
    void setGameStatus(boolean started);
    void clearGameSteps();

}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/players/PlayersListInteractor.kt
package com.datarockets.mnchkn.activities.players

import com.datarockets.mnchkn.models.Player

interface PlayersListInteractor {

    interface OnFinishedListener {
        fun onPlayersLoaded(players: List<Player>)
        fun onPlayerAdded(player: Player)
        fun onPlayerDeleted(position: Int)
        fun onPlayersCountChecked(enough: Boolean)
        fun onGameStarted(started: Boolean)
    }

    fun clearPlayersStats(listener: OnFinishedListener)
    fun isGameStarted(listener: OnFinishedListener)
    fun checkIsEnoughPlayer(listener: OnFinishedListener)
    fun getPlayers(listener: OnFinishedListener)
    fun addPlayer(name: String, listener: OnFinishedListener)
    fun deletePlayer(position: Int, id: Long, listener: OnFinishedListener)
    fun setGameStatus(started: Boolean)
    fun clearGameSteps()

}
